File: bsp4rs/src/com/jetbrains/bsp/bsp4rs/bsp/build_target_event.rs
use serde::{Deserialize, Serialize};

use crate::*;

#[derive(Clone, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct BuildTargetEvent {
    /** The identifier for the changed build target */
    #[serde(default)]
    pub target: BuildTargetIdentifier,
    /** The kind of change for this build target */
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub kind: Option<BuildTargetEventKind>,
    /** Any additional metadata about what information changed. */
    #[serde(default, flatten, skip_serializing_if = "Option::is_none")]
    pub data: Option<BuildTargetEventData>,
}

File: bsp4kt/src/org/jetbrains/bsp/bsp4kt/BuildTargetEvent.kt
package org.jetbrains.bsp.bsp4kt

import kotlinx.serialization.Serializable
import kotlinx.serialization.SerialName
import kotlinx.serialization.json.JsonElement

@Serializable
data class BuildTargetEvent(
  val target: BuildTargetIdentifier,
  val kind: BuildTargetEventKind? = null,
  val data: BuildTargetEventData? = null
)
